modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -2143,4 +2166,40 @@ public boolean isWorkflowActive(String workflowState) {
             || WorkflowState.PAUSED.toString().equals(workflowState);
   }
 
+  private void checkAcceptedTypes(String assetUploadId, MediaType mediaType) {
+    if (mediaType.is(MediaType.OCTET_STREAM)) {
+      // No detailed info, so we have to accept...
+      return;
+    }
+    final JSONParser parser = new JSONParser();
+    try {
+      final Collection<String> assetUploadJsons = listProvidersService.getList("eventUploadAssetOptions",
+          new ResourceListQueryImpl(),false).values();
+      for (String assetUploadJson: assetUploadJsons) {
+        if (!assetUploadJson.startsWith("{") || !assetUploadJson.endsWith("}")) {
+          // Ignore non-json-values
+          continue;
+        }
+        @SuppressWarnings("unchecked")
+        final Map<String, String> assetUpload = (Map<String, String>) parser.parse(assetUploadJson);
+        if (assetUploadId.equals(assetUpload.get("id"))) {
+          final List<String> accepts = Arrays.stream(assetUpload.getOrDefault("accept", "*/*").split(","))
+              .map(String::trim).collect(Collectors.toList());
+          for (String accept : accepts) {
+            if (accept.contains("/") && mediaType.is(MediaType.parse(accept))) {
+              return;
+            } else if (mediaType.subtype().equalsIgnoreCase(accept.replace(".", ""))) {
+              return;
+            }
+          }
+          throw new IllegalArgumentException("Provided file format " + mediaType.toString() + " not allowed.");
```

Comment:
I agree that throwing the exceptions inside this method is a bit counter-intuitive. However, returning a boolean here would not be sufficient since I wouldn't be able to throw an excpetion with a precise message when evaluating the boolean. This method would have to return an enum instead which would then have to be evaluated in a big switch-case. And since we would have to do this in two places, I think that would be a lot of code overhead. So i would rather stick with throwing inside the method which performs the check.